DQM/HcalTasks/plugins/ZDCQIE10Task.cc

```diff
@@ -1,116 +1,579 @@
+  _cZDC_HAD_TM[0]->setBinLabel(5, "HAD_M_3 good timing");
+  _cZDC_HAD_TM[0]->setBinLabel(6, "HAD_M_3 bad timing");
+  _cZDC_HAD_TM[0]->setBinLabel(7, "HAD_M_4 good timing");
+  _cZDC_HAD_TM[0]->setBinLabel(8, "HAD_M_4 bad timing");
+
+  histoname = "HAD_P_Timings";
+  ib.setCurrentFolder("Hcal/ZDCQIE10Task");
+  _cZDC_HAD_TM[1] = ib.book1DD(histoname.c_str(), histoname.c_str(), 8, 1, 9);
+  _cZDC_HAD_TM[1]->setAxisTitle("Channel", 1);
+  _cZDC_HAD_TM[1]->setAxisTitle("N", 2);
+  _cZDC_HAD_TM[1]->setBinLabel(1, "HAD_P_1 good timing");
+  _cZDC_HAD_TM[1]->setBinLabel(2, "HAD_P_1 bad timing");
+  _cZDC_HAD_TM[1]->setBinLabel(3, "HAD_P_2 good timing");
+  _cZDC_HAD_TM[1]->setBinLabel(4, "HAD_P_2 bad timing");
+  _cZDC_HAD_TM[1]->setBinLabel(5, "HAD_P_3 good timing");
+  _cZDC_HAD_TM[1]->setBinLabel(6, "HAD_P_3 bad timing");
+  _cZDC_HAD_TM[1]->setBinLabel(7, "HAD_P_4 good timing");
+  _cZDC_HAD_TM[1]->setBinLabel(8, "HAD_P_4 bad timing");
+
+  //book histos per channel
+  //std::string histoname;
   for (int channel = 1; channel < 6; channel++) {
     // EM Pos
     HcalZDCDetId didp(HcalZDCDetId::EM, true, channel);
-    histoname = "EM_P_" + std::to_string(channel) + "_1";
+    // cout << " ZDC EM P eid " <<didp()<< " ch "<<channel<<endl;
+
+    histoname = "EM_P_" + std::to_string(channel);
     ib.setCurrentFolder("Hcal/ZDCQIE10Task/ADC_perChannel");
     _cADC_EChannel[didp()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 256, 0, 256);
     _cADC_EChannel[didp()]->setAxisTitle("ADC", 1);
     _cADC_EChannel[didp()]->setAxisTitle("N", 2);
     ib.setCurrentFolder("Hcal/ZDCQIE10Task/ADC_vs_TS_perChannel");
-    _cADC_vs_TS_EChannel[didp()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 10, 0, 10);
+    _cADC_vs_TS_EChannel[didp()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 6, 0, 6);
     _cADC_vs_TS_EChannel[didp()]->setAxisTitle("TS", 1);
     _cADC_vs_TS_EChannel[didp()]->setAxisTitle("sum ADC", 2);
 
+    histoname = "EM_P_" + std::to_string(channel);
+    ib.setCurrentFolder("Hcal/ZDCQIE10Task/fC_perChannel");
+    _cfC_EChannel[didp()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 100, 0, 8000);
+    _cfC_EChannel[didp()]->setAxisTitle("fC", 1);
+    _cfC_EChannel[didp()]->setAxisTitle("N", 2);
+    ib.setCurrentFolder("Hcal/ZDCQIE10Task/fC_vs_TS_perChannel");
+    _cfC_vs_TS_EChannel[didp()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 6, 0, 6);
+    _cfC_vs_TS_EChannel[didp()]->setAxisTitle("TS", 1);
+    _cfC_vs_TS_EChannel[didp()]->setAxisTitle("sum fC", 2);
+
+    histoname = "EM_P_" + std::to_string(channel);
+    _cTDC_EChannel[didp()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 50, 1, 50);
+    _cTDC_EChannel[didp()]->setAxisTitle("TDC", 1);
+    _cTDC_EChannel[didp()]->setAxisTitle("N", 2);
+
     // EM Minus
     HcalZDCDetId didm(HcalZDCDetId::EM, false, channel);
-    histoname = "EM_M_" + std::to_string(channel) + "_1";
+
+    //cout << " ZDC EM M eid " <<didm()<< " ch "<<channel<<endl;
```

Comment:
As in previous comment, remove commented line

Reply:
This was removed as part of https://github.com/cms-sw/cmssw/commit/eff9cc2e664dc4e8381d0140a0c4dc94965ee6d5